src/main/java/codeu/bot/ActionMatcher.java

```diff
@@ -181,154 +224,212 @@ private void matchSendMessage() throws IOException {
             Matcher messageMatcher = messagePattern.matcher(input);
             if (messageMatcher.find()) {
                 message = messageMatcher.group(1);
+                String conversationTitle = findFuzzyMatch(conversationTitles, 5);
+                System.out.printf("Sending message \"%s\" in conversation <%s>.",message,conversationTitle);
+                BotActions.Action.SEND_MESSAGE.doAction(message,conversationTitle);
+                actionMatched = true;
+                return;
+            } else {
+                BotActions.Action.MISSING_PARAMETER.doAction();
+                actionMatched = true;
+                return;
             }
-
-            String conversationTitle = findFuzzyMatch(conversationTitles, 5);
-            System.out.printf("Sending message \"%s\" in conversation <%s>.",message,conversationTitle);
-            BotActions.Action.SEND_MESSAGE.doAction(message,conversationTitle);
-            return;
         }
+        return;
     }
 
-    public void matchGet() throws IOException {
-        int getOrFindOrDisplayorShoworGiveIndex = getKeywordIndex(GET_KEYWORDS, commandTokensLemmas);
-        if (getOrFindOrDisplayorShoworGiveIndex != -1 && 
-            commandPOS.get(getOrFindOrDisplayorShoworGiveIndex).contains("VB")) {
+    public void matchGet() throws IOException, PersistentDataStoreException {
+        int getOrFindOrDisplayOrShoworGiveIndex = getKeywordIndex(GET_KEYWORDS, commandTokensLemmas);
+        if (getOrFindOrDisplayOrShoworGiveIndex != -1
+            // && commandPOS.get(getOrFindOrDisplayOrShoworGiveIndex).contains("VB")
+            ) {
 
             // Getting the lemma so we can ignore pluralizations
-            IndexedWord verbOfInterest = new IndexedWord(commandTokens.get(getOrFindOrDisplayorShoworGiveIndex));
+            IndexedWord verbOfInterest = new IndexedWord(commandTokens.get(getOrFindOrDisplayOrShoworGiveIndex));
         
             // Find object we're getting
             Set<IndexedWord> directObjects = dependencyParse.getChildrenWithReln(verbOfInterest,UniversalEnglishGrammaticalRelations.DIRECT_OBJECT);
             String directObject = "";
-            if (!directObjects.isEmpty() && directObjects.size() == 1) {
+
+            if (directObjects.size() == 1) {
                 directObject = (new ArrayList<>(directObjects)).get(0).lemma();
             } else {
-                // Make a dummy call to SET_DESCRIPTION
-                BotActions.Action.SET_DESCRIPTION.doAction();
                 return;
             }
 
+            System.out.println("here"+directObject);
+
             switch (directObject) {
                 case "message": {
                     matchGetMessages();
+                    break;
                 }
                 case "conversation": {
                     matchGetConversations();
+                    break;
+                }
+            }
+
+            if (actionMatched) {
+                return;
```

Comment:
Since you return here you dont need an "else" statement

Reply:
Fixed.